single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/layout/positionfromextension/PositionFromExtension.java

```diff
@@ -117,17 +117,26 @@ public LBSCluster organizeLegBusSets(VoltageLevelGraph graph, List<LegBusSet> le
     }
 
     private void gatherLayoutExtensionInformation(VoltageLevelGraph graph) {
-        graph.getNodes().stream().filter(node -> node.getDirection() != Direction.UNDEFINED && graph.getCell(node).isPresent()).forEach(node -> {
-            BusCell cell = (BusCell) graph.getCell(node).get();
-            cell.setDirection(node.getDirection());
-            node.getOrder().ifPresent(cell::setOrder);
-        });
+
         graph.getBusCellStream().forEach(bc -> {
+            List<Direction> listOfDirectionsInsideCell = bc.getNodes().stream().map(Node::getDirection).distinct().collect(Collectors.toList());
+            int numberOfDirectionsInsideCell = listOfDirectionsInsideCell.size();
+            if (numberOfDirectionsInsideCell == 1) {
+                Direction direction = listOfDirectionsInsideCell.get(0);
+                if (direction != Direction.UNDEFINED) {
+                    bc.setDirection(direction);
+                }
+            } else if (numberOfDirectionsInsideCell > 1) {
+                bc.setDirection(listOfDirectionsInsideCell.get(numberOfDirectionsInsideCell - 1));
```

Comment:
I'd say `get(0)` instead, or is there any reason to choose last one rather than first one?

Reply:
I chose the last one so that, if we ever fall into this case, we would not differ from what was done before.
But if you think it is more logical to get the first one, it is fine for me too !